File: Python-Basics/loops/exam_preparation.py
failed_marks = int(input())

task = input()
sum_grades = 0.0
passed_tasks = 0
failed_tasks = 0
last_task = ''

while task != 'Enough':
    grade = float(input())
    sum_grades += grade

    if grade > 4.0:
        passed_tasks += 1
    else:
        failed_tasks += 1

    last_task = task

    if failed_tasks == failed_marks:
        print(f'You need a break, {failed_marks} poor grades.')
        break

    task = input()

if failed_tasks != failed_marks:
    print(f'Average score: {sum_grades/(passed_tasks+failed_tasks):.2f}')
    print(f'Number of problems: {passed_tasks+failed_tasks}')
    print(f'Last problem: {last_task}')

File: CPP-Basics/Exam Preparation.cpp
#include <iostream>
#include <string>

using namespace std;

int main () {

    int maxFails;
    cin >> maxFails;
    cin.ignore();

    string input;
    getline(cin, input);


    int failCount = 0;
    int sumGrades = 0;
    int countTasks = 0;
    string lastProblem = "";

    while(input != "Enough") {

        string taskName = input;
        int grade;

        cin >> grade;
        cin.ignore();

        sumGrades += grade;
        countTasks ++;

        if(grade <= 4) {

            failCount++;
        }

        if(failCount == maxFails) {

            cout << "You need a break, " << failCount << " poor grades." << endl;
            break;
        }

        lastProblem = taskName;

        getline(cin,input);

    }

    if(input == "Enough") {

        double average = sumGrades * 1.0 / countTasks;
        cout.setf(ios::fixed);
        cout.precision(2);
        cout << "Average score: " << average << endl;
        cout << "Number of problems: " << countTasks << endl;
        cout << "Last problem: " << lastProblem << endl;
    }

    return 0;
}
